### PR TITLE
Api doc + dtos + Swagger of course completed with other fixes

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
Api doc + dtos + Swagger of course completed with other fixes

npm run lint:fix
npm run test
nest start 
tests run to check changes

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] My changes generate no new warnings
- [x] The title of my pull request is a short description of the requested changes.

Swagger has been successfully generated along with other fixes

### CODE REVIEW

File: src/modules/assignment/assignment.service.ts [hunk 1]
```diff
@@ -14,16 +19,37 @@ export class AssignmentService {
...
-    return Assignments;
+    try {
+      const Assignments = await this.AssignmentModel.find().exec();
+      return Assignments;
+    } catch (e) {
+      throw new HttpException(
+        {
+          status: HttpStatus.INTERNAL_SERVER_ERROR,
+          error: `${e}`,
+        },
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.service.ts:30]: Please use the correct Methods

File: src/modules/assignment/assignment.service.ts [hunk 2]
```diff
@@ -39,28 +65,68 @@ export class AssignmentService {
...
   ): Promise<Assignment> {
-    const newAssignment = await new this.AssignmentModel(CreateAssignmentDTO);
-    return newAssignment.save();
+    try {
+      const newAssignment = await new this.AssignmentModel(CreateAssignmentDTO);
+      await newAssignment.save();
+      if (newAssignment) {
+        return newAssignment;
+      }
+      throw new NotFoundException('doubt not found!');
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.service.ts:74]: No need for this, just return the saved assignments

File: src/modules/assignment/assignment.service.ts [hunk 3]
```diff
@@ -39,28 +65,68 @@ export class AssignmentService {
...
+    } catch (e) {
+      throw new HttpException(
+        {
+          status: HttpStatus.INTERNAL_SERVER_ERROR,
+          error: `${e}`,
+        },
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+    throw new NotFoundException('doubt not found!');
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.service.ts:109]: Already told you to replace the doubts with assignments :(

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: src/modules/assignment/assignment.service.ts (hunk 3 above)
Comment (Devesh21700Kumar) [reply src/modules/assignment/assignment.service.ts:109]: this is a different branch
PE#90 has the changes made

File: src/modules/assignment/assignment.service.ts (hunk 2 above)
Comment (Devesh21700Kumar) [reply src/modules/assignment/assignment.service.ts:74]: this is a different branch
PR#90 has the changes made

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 do checkout this course swagger doc PR

File: src/modules/course/docUtils/course.responsedoc.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+   * The name of the course
+   * @example 'Rest Apis'
+   */
+  name: string;
+
+  /**
+   * The price of the course
+   * @example 400INR
+   */
+  price: number;
```
Comment (Abhishek-kumar09) [inline src/modules/course/docUtils/course.responsedoc.ts:20]: price is accepting number so why there is INR in example.

File: src/modules/course/docUtils/course.responsedoc.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+   * The end date of the course
+   * @example '2020-02-05T06:35:22.000Z'
+   */
+  end_date: Date;
+
+  /**
+   * The shareable link of the course
+   * @example '88900xyz.com'
+   */
+  sharable_link: string;
```
Comment (Abhishek-kumar09) [inline src/modules/course/docUtils/course.responsedoc.ts:68]: Example of links should start with https://
eg..`'https://sharable_link.com'`

Comment (Abhishek-kumar09) [reply src/modules/course/docUtils/course.responsedoc.ts:68]: spelling of sharable

File: src/modules/course/docUtils/course.responsedoc.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+   */
+
+  chapterName: string;
+
+  /**
+   * The description of the schedule item
+   * @example 'Rest Apis'
+   */
+
+  description: string;
```
Comment (Abhishek-kumar09) [inline src/modules/course/docUtils/course.responsedoc.ts:102]: Examples are meant to be ideal representation from software creators.
Please provide the better examples that if somneone reads could understand the motive.

File: src/modules/course/docUtils/course.responsedoc.ts [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+  /**
+   * The time required to finish the scheduled item
+   * @example 'Rest Apis'
+   */
+
+  time: string;
+}
```
Comment (Abhishek-kumar09) [inline src/modules/course/docUtils/course.responsedoc.ts:110]: Time's example is RestAPI :confused:

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: src/modules/course/docUtils/course.responsedoc.ts (hunk 5 above)
Comment (Devesh21700Kumar) [reply src/modules/course/docUtils/course.responsedoc.ts:68]: i guess the spelling is shareable only.
I'll change the link

File: src/modules/course/docUtils/course.responsedoc.ts (hunk 7 above)
Comment (Devesh21700Kumar) [reply src/modules/course/docUtils/course.responsedoc.ts:110]: LOL.. will change sorry.
missed it out

File: src/modules/course/docUtils/course.responsedoc.ts (hunk 6 above)
Comment (Devesh21700Kumar) [reply src/modules/course/docUtils/course.responsedoc.ts:102]: so what should i write  here?
somwhat like...
A session on REST API which  is an application programming interface (API or web API) that conforms to the constraints of REST architectural style and allows for interaction with RESTful web services.

**Review by Devesh21700Kumar — COMMENTED** [review]
Examples have been fixed. 
@Abhishek-kumar09 pls check

**Review by Abhishek-kumar09 — APPROVED** [review]

**Pull request merged**